op-acceptance-tests/tests/base/reth_geth_divergence_test.go

```diff
@@ -0,0 +1,204 @@
+package base
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/system"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/testing/systest"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+// TestRethGethDivergence checks that reth and geth do not diverge in terms of state root.
+// geth is considered the canonical/correct implementation.
+func TestRethGethDivergence(t *testing.T) {
+	systest.SystemTest(t, rethGethDivergenceTestScenario())
+}
+
+func rethGethDivergenceTestScenario() systest.SystemTestFunc {
+	const (
+		// testTimeout is the maximum time allowed for the test to run
+		testTimeout = 60 * time.Second
+
+		// blockCheckCount is the number of recent blocks to check for divergence
+		blockCheckCount = 5
```

Comment:
Is this far back enough, or should we have a time-based check? For example, check for divergence for 2 minutes.